### PR TITLE
Create some util funcs that are common between start and exec

### PR DESCRIPTION
**Pull request by duglin**
and it'll really help my start/create PR when I need to rebase  :-)

Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

**Comment by crosbymichael** [comment]
LGTM

**Comment by duglin** [comment]
ping @mrunalp @LK4D4

**Comment by duglin** [comment]
rebased

**Comment by mrunalp** [comment]
Needs rebase because of the other merges.

**Comment by duglin** [comment]
@mrunalp rebased

File: utils.go [hunk 1]
```diff
@@ -265,3 +265,64 @@ func createPidFile(path string, process *libcontainer.Process) error {
...
+
+func createContainer(context *cli.Context, id string, spec *specs.LinuxSpec) (libcontainer.Container, error) {
+	config, err := createLibcontainerConfig(id, spec)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := os.Stat(config.Rootfs); err != nil {
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf("Rootfs (%q) does not exist", config.Rootfs)
```
Comment (mrunalp) [inline utils.go]: nit: Rootfs -> rootfs (start error strings with lowercase)

Comment (duglin) [inline utils.go]: fixed

File: utils.go [hunk 2]
```diff
@@ -265,3 +265,64 @@ func createPidFile(path string, process *libcontainer.Process) error {
...
+	}
+
+	factory, err := loadFactory(context)
+	if err != nil {
+		return nil, err
+	}
+	return factory.Create(id, config)
+}
+
+// runProcess will create a new process (PID ns) in the specified container
```
Comment (mrunalp) [inline utils.go]: This comment isn't accurate as all the namespaces are created in init today.

Comment (duglin) [inline utils.go]: @mrunalp updated

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**